### DISCUSSION TITLE
An exception occurred

### DISCUSSION THREAD

**Discussion by betoxx** [aldas/modbus-tcp-client#130] [Q&A]
Hi, I'm trying to connect to a Siemens Logo with index.php in the examples folder but I receive this message:

Using: function code: 3, ip: 192.168.1.199, port: 502, address: 0, quantity: 12, endianess: 1
Packet to be sent (in hex): 3e170000000600030000000c
An exception occurred
fread error during receiveFrom
#0 C:\xampp\htdocs\2\vendor\aldas\modbus-tcp-client\src\Network\BinaryStreamConnection.php(60): ModbusTcpClient\Network\BinaryStreamConnection->receiveFrom(Array, 10, NULL)
#1 C:\xampp\htdocs\2\vendor\aldas\modbus-tcp-client\src\Network\BinaryStreamConnection.php(83): ModbusTcpClient\Network\BinaryStreamConnection->receive()
#2 C:\xampp\htdocs\2\vendor\aldas\modbus-tcp-client\examples\index.php(54): ModbusTcpClient\Network\BinaryStreamConnection->sendAndReceive(Object(ModbusTcpClient\Packet\ModbusFunction\ReadHoldingRegistersRequest))
#3 {main}

I had to change the path to autoload.php on the first line to:
require 'C:\xampp\htdocs\2\vendor\autoload.php';

But i think that this is not the problem. Could someone help me on this please?

Thank you!

**Comment by aldas** [comment]
Could you verify first with https://github.com/aldas/modbus-tcp-client/wiki/Debugging-guide that other clients work with same parameters (IP/port/unitid)

another thing to check - is your PLC  using ModbusTCP or ModbusRTU over tcp?

**Comment by betoxx** [comment]
The PLC is using ModbusTCP.
I have found a solution deleting inputs not used in the PLC and Ethernet Connections opened by the PLC. There was 8, with the ports 502 to 510. Now I left only one to 502. I will paste some screnshots and the results given by ../aldas/modbus-tcp-client/examples/index.php
![screen1](https://user-images.githubusercontent.com/6755011/215982403-844a6800-8cd1-4c61-a7d7-38971ab204a5.JPG)
![screen2](https://user-images.githubusercontent.com/6755011/215982413-1e467b09-af4e-483b-8e42-66ecccbd06be.JPG)
![screen3](https://user-images.githubusercontent.com/6755011/215982414-bd8a253e-47f6-4fbb-be87-b78c3db1ab69.JPG)
![screen4](https://user-images.githubusercontent.com/6755011/215982419-3da09607-0fe1-47e7-8a08-6822fbb7adf7.JPG)
![screen5](https://user-images.githubusercontent.com/6755011/215982422-3e50bbf9-ad07-4b08-a91b-da4c5adcc92c.JPG)

**Comment by betoxx** [comment]
Thank you for your sugest!!!!!
